Pcap++/src/DpdkDevice.cpp

```diff
@@ -289,12 +289,15 @@ bool DpdkDevice::configurePort(uint8_t numOfRxQueues, uint8_t numOfTxQueues)
 		return false;
 	}
 
-	// verify num of RX queues is power of 2
-	bool isRxQueuePowerOfTwo = !(numOfRxQueues == 0) && !(numOfRxQueues & (numOfRxQueues - 1));
-	if (!isRxQueuePowerOfTwo)
+	// verify num of RX queues is power of 2 for virtual devices
+	if (isVirtual())
 	{
-		PCPP_LOG_ERROR("Num of RX queues must be power of 2 (because of DPDK limitation). Attempted to open device with " << numOfRxQueues << " RX queues");
-		return false;
+		bool isRxQueuePowerOfTwo = !(numOfRxQueues == 0) && !(numOfRxQueues & (numOfRxQueues - 1));
```

Comment:
let's refactor this: `!(numOfRxQueues == 0)` -> (numOfRxQueues != 0)

Reply:
We can also do:
```cpp
bool isRxQueuePowerOfTwo = !(numOfRxQueues == 0 || (numOfRxQueues & (numOfRxQueues - 1)));
```

I'm not sure what is nicer though...

Reply:
Perhaps the check for power of 2 and the number of queues being zero should be separated? We should throw an error when the requested number of queues is zero, right? Even in the non-virtual case

Reply:
@lucashc you're right, I totally agree! 👍

Reply:
Actually, do we support only sending packets? In that case it might make sense to allow zero rx queues. 
In any case, I will add the patch for separating the checks.

Reply:
we could support only sending packets, but I think this is a very uncommon use-case. I think we can keep the restriction to disallow zero rx queues